numba/cuda/cudaimpl.py

```diff
@@ -245,8 +245,15 @@ def ptx_threadfence_device(context, builder, sig, args):
     return context.get_dummy_value()
 
 
-@lower(stubs.syncwarp, types.i4)
-def ptx_warp_sync(context, builder, sig, args):
+@lower(stubs.syncwarp)
+def ptx_syncwarp(context, builder, sig, args):
+    mask = context.get_constant(types.uint32, 0xFFFFFFFF)
+    mask_sig = signature(types.none, types.uint32)
+    return ptx_syncwarp_mask(context, builder, mask_sig, [mask])
+
+
+@lower(stubs.syncwarp, types.u4)
```

Comment:
Why the removal of the signed mask?

Reply:
Because its argument is a bitmask, I felt that it's more natural for the argument to be unsigned, and that passing a signed value might be indicative of a logic error in the user's code.

This also makes Numba more consistent with CUDA C/C++, where the argument to `__syncwarp` is unsigned: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#synchronization-functions

Reply:
I entirely agree that bitmasks are naturally unsigned, just wonder if such a change might brake in prod user code that assumes signed?

Reply:
I haven't been able to work out a scenario where it would, but to be conservative I've changed it back to signed. If the masks are to change to unsigned (which I would like at some point in the future) then it makes more sense to change them all at once, and not just for one intrinsic, along with some careful testing.